pkg/sql/catalog/bootstrap/metadata.go

```diff
@@ -115,21 +115,19 @@ func (ms MetadataSchema) GetInitialValues() ([]roachpb.KeyValue, []roachpb.RKey)
 				Key:   catalogkeys.MakePublicObjectNameKey(ms.codec, parentID, desc.GetName()),
 				Value: value,
 			})
+		} else if desc.GetID() == keys.PublicSchemaID {
+			ret = append(ret, roachpb.KeyValue{
+				Key:   catalogkeys.MakePublicSchemaNameKey(ms.codec, parentID),
+				Value: value,
+			})
 		} else {
-			// Initializing a database. Databases must be initialized with
-			// the public schema, as all tables are scoped under the public schema.
-			publicSchemaValue := roachpb.Value{}
-			publicSchemaValue.SetInt(int64(keys.PublicSchemaID))
 			ret = append(
 				ret,
 				roachpb.KeyValue{
 					Key:   catalogkeys.MakeDatabaseNameKey(ms.codec, desc.GetName()),
 					Value: value,
 				},
-				roachpb.KeyValue{
-					Key:   catalogkeys.MakePublicSchemaNameKey(ms.codec, desc.GetID()),
-					Value: publicSchemaValue,
-				})
+			)
```

Comment:
This whole if/else block can be replaced with
```
ret = append(ret, roachpb.KeyValue{Key: catalogkeys.EncodeNameKey(ms.codec, desc), Value: value})
```
as we now have a 1-to-1 mapping from descriptor to namespace entry.